src/main/java/com/coreos/jetcd/EtcdClient.java

```diff
@@ -33,4 +67,46 @@ public EtcdClient(ManagedChannelBuilder<?> channelBuilder, EtcdClientBuilder bui
     public EtcdKV newKVClient(KVGrpc.KVFutureStub stub) {
         return new EtcdKVImpl(stub);
     }
+
+    private EtcdAuth newAuthClient(AuthGrpc.AuthFutureStub stub){
+        return new EtcdAuthImpl(stub);
+    }
+
+    protected EtcdAuth getAuthClient(){
+        return authClient;
+    }
+
+    protected EtcdKV getKVClient(){
+        return kvClient;
+    }
+
+    /**
+     * get token from etcd with name and password
+     *
+     * @param channel channel to etcd
+     * @param name auth name
+     * @param password auth password
+     * @return token string
+     */
+    private String getToken(ManagedChannel channel, ByteString name, ByteString password) {
+
+        try {
+            ListenableFuture<AuthenticateResponse> authResp = AuthGrpc.newFutureStub(channel)
+                    .authenticate(
+                            AuthenticateRequest.newBuilder()
+                                    .setNameBytes(name)
+                                    .setPasswordBytes(password)
+                                    .build());
+            return authResp.get().getToken();
+        }catch (InterruptedException ie){
+            return null;
```

Comment:
password might be wrong. user should fill in the correct one. connection might be bad, need to retry.